Orange/base.py

```diff
@@ -262,13 +294,14 @@ def fit(self, X, Y, W=None):
             return self.__returns__(clf.fit(X, Y))
         return self.__returns__(clf.fit(X, Y, sample_weight=W.reshape(-1)))
 
-    def __str__(self):
-        return '{} {}'.format(self.name, self.params)
-
     def __repr__(self):
-        return '{}({})'.format(type(self).__name__,
-                               ", ".join("{}={}".format(k, v)
-                                         for k, v in self.params.items()))
+        # In addition to saving values onto self, SklLearners save into params
+        vars = self.__dict__
+        self.__dict__ = dict(vars, **self.params)
```

Comment:
This looks alot like patching the self.__dict__. Any specific reason why not using [patch.object](https://docs.python.org/3/library/unittest.mock.html#patch-object)?